R/SAFFRONstar.R

```diff
@@ -107,42 +110,63 @@
 #'
 #'
 #' @examples
-#' pval = c(2.90e-08, 0.06743, 3.51e-04, 0.0174, 0.04723,
-#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-06,
-#'         0.69274, 0.30443, 0.00136, 0.82342, 0.54757)
-#'
-#' SAFFRONstar(pval, version='async', decision.times=seq_len(15)) # Synchronous
+#' sample.df <- data.frame(
+#' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757),
+#' decision.times = seq_len(15) + 1)
+#'
+#' SAFFRONstar(sample.df, version='async')
 #' 
-#' SAFFRONstar(pval, version='async', decision.times=seq_len(15)+1)
-#' # Asynchronous
-#'
-#'
-#' SAFFRONstar(pval, version='dep', lags=rep(0,15)) # Synchronous
+#' sample.df2 <- data.frame(
+#' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757),
+#' lags = rep(1,15))
 #' 
-#' SAFFRONstar(pval, version='dep', lags=rep(1,15)) # Locally dependent
-#'
-#'
-#' SAFFRONstar(pval, version='batch', batch.size=rep(1,15)) # Synchronous
+#' SAFFRONstar(sample.df2, version='dep')
 #' 
-#' SAFFRONstar(pval, version='batch', batch.size=c(4,6,5)) # Batched
+#' sample.df3 <- data.frame(
+#' #' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757))
+#' 
+#' SAFFRONstar(sample.df3, version='batch', batch.sizes = c(4,6,5))
 #'
 #' @export
 
-SAFFRONstar <- function(pval, alpha=0.05, version, gammai, w0, lambda=0.5,
-                        decision.times, lags, batch.sizes,
+SAFFRONstar <- function(d, alpha=0.05, version, gammai, w0, lambda=0.5, batch.sizes,
                         discard=FALSE, tau.discard=0.5) {
-
-    if(alpha<=0 || alpha>1){
-        stop("alpha must be between 0 and 1.")
-    }
-
-    if(lambda<=0 || lambda>1){
-        stop("lambda must be between 0 and 1.")
-    }
-    
-    if(version == 'async' && discard){
-        return(ADDIS(pval, alpha, gammai, w0, lambda, tau.discard,
-                 async=TRUE, decision.times))
+  
+  if(is.data.frame(d)){
+    checkSTARdf(d, version)
+    pval <- d$pval
+  } else if(is.vector(d)){
+    pval <- d
```

Comment:
Add the following check: if version = 'async' or 'dep', then d should be a dataframe with a column of decision.times or lags respectively

Reply:
Done